python/cugraph/cugraph/structure/property_graph.py

```diff
@@ -784,6 +784,27 @@ def get_edge_data(self, edge_ids=None, types=None, columns=None):
 
         return None
 
+    def fillna(self, val=0, props_to_fill="both"):
+        """
+        Fills empty property values with the given value, zero by default.
+        Can fill vertex properties, edge properties or both (default).
+        Fills in-place.
+
+        Parameters
+        ----------
+        val : object
+            The object that will replace "na". Default = 0
+        props_to_fill : 'vertex', 'edge', or 'both' (default)
+            Whether to fill vertex properties only, edge properties only,
+            or both vertex and edge properties (default).
+        """
+        if props_to_fill == "vertex" or props_to_fill == "both":
+            for prop in self.vertex_property_names:
+                self.__vertex_prop_dataframe[prop].fillna(val, inplace=True)
```

Comment:
I don't think `inplace=True` here will reliably work with pandas.  One can pass a Series or dict to `fillna`, so we could do e.g.
```python
df.fillna(dict.fromkeys(self.vertex_property_names, val), inplace=True)
```
Should we consider accepting (or checking for) a dict, series, or even dataframe as the fill value?

@alexbarghi-nv let me know if you'd like me to take over implementing this feature so we can be sure to get it in 22-12.

Reply:
I will allow dict/series/dataframe as the fill value.  I think that also removes the need for a `columns` kwarg.

Reply:
I updated it, and will update the tests as well, but I'm just worried about the MG implementation running out of memory.  Do you have a proposal to do something sort-of in place for MG, or do you think dask will be able to handle things at our current scale for now?

Reply:
Yeah, I think dask will be able to handle things, so I wouldn't worry about trying to optimize this further. If we try to "hack" MG to do things in-place, we lose resilience.